go/cmd/dolt/commands/checkout.go

```diff
@@ -144,6 +142,74 @@ func (cmd CheckoutCmd) Exec(ctx context.Context, commandStr string, args []strin
 	return HandleVErrAndExitCode(verr, usagePrt)
 }
 
+func checkoutNewBranch(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults) errhand.VerboseError {
+	var newBranchName string
+	var remoteName string
+	var remoteBranchName string
+	var startPt = "head"
+
+	if apr.NArg() == 1 {
+		startPt = apr.Arg(0)
+	}
+
+	trackVal, setTrackUpstream := apr.GetValue(cli.TrackFlag)
+	if setTrackUpstream {
+		if trackVal != "direct" && trackVal != "inherit" {
+			startPt = trackVal
+		} else if trackVal == "inherit" {
+			return errhand.VerboseErrorFromError(fmt.Errorf("--track='inherit' is not supported yet"))
+		}
+		remoteName, remoteBranchName = parseRemoteBranchName(startPt)
+		remotes, err := dEnv.RepoStateReader().GetRemotes()
+		if err != nil {
+			return errhand.BuildDError(err.Error()).Build()
+		}
+		_, remoteOk := remotes[remoteName]
+		if !remoteOk {
+			return errhand.BuildDError(fmt.Errorf("'%s' is not a commit and a branch '%s' cannot be created from it", startPt, remoteBranchName).Error()).Build()
```

Comment:
Does this match the Git error message? "is not a commit" seemed a little confusing to me, since we know this should be a remote/branch in this block, since the track flag is set. I'm wondering if "is not a valid remote ref" or something.